aspnet/security/anti-request-forgery.md

```diff
@@ -0,0 +1,85 @@
+---
+title: Preventing Cross-Site Request Forgery (XSRF/CSRF) Attacks in ASP.NET Core
+author: rick-anderson
+ms.author: riande
+manager: wpickett
+ms.date: 10/14/2016
+ms.topic: article
+ms.assetid: 09607e8b-a977-4807-81aa-7a5d73bfbacd
+ms.prod: aspnet-core
+uid: security/anti-request-forgery
+---
+
+  # Preventing Cross-Site Request Forgery (XSRF/CSRF) Attacks in ASP.NET Core
+
+[Fiyaz Hasan](https://twitter.com/FiyazBinHasan), David Paquette
+
+See this [word document](https://www.dropbox.com/sh/jfmncpp0z79m9mt/AAAAB98zC4J7f9g_l6egU8nta?dl=0) and [Evolving ASP.NET Apps–Cookie Authentication](https://blogs.msdn.microsoft.com/cdndevs/2015/02/18/evolving-asp-net-appscookie-authentication/)
+
+David Paquette gave me permission to copy his blog if we add him to the authors list.
+
+Cross-site request forgery (also known as XSRF or CSRF) is an attack against web-hosted applications whereby a malicious web site can influence the interaction between a client browser and a web site trusted by that browser. These attacks are made possible because web browsers send authentication tokens automatically with every request to a web site.
+
+Here is an example of a CSRF attack:
+
+1. A user logs into `www.example.com`, using forms authentication.
+
+2. The server authenticates the user and issues a  response that includes an authentication cookie.
+
+3. The attacker uses social engineering to trick the user into clicking on a link to a malacious site.
+
+   This malicious site contains the following HTML form:
+
+<!-- literal_block {"xml:space": "preserve", "language": "html", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+````html
+
+   <h1>You Are a Winner!</h1>
+     <form action="http://example.com/api/account" method="post">
+       <input type="hidden" name="Transaction" value="withdraw" />
+       <input type="hidden" name="Amount" value="1000000" />
+     <input type="submit" value="Click Me"/>
+   </form>
+   ````
+
+Notice that the form action posts to the vulnerable site, not to the malicious site. This is the “cross-site” part of CSRF.
+
+4. The user clicks the submit button. The browser includes the authentication cookie with the request.
+
+5. The request runs on the server with the user’s authentication context, and can do anything that an authenticated user is allowed to do.
+
+Although this example requires the user to click the form button, the malicious page could just as easily run a script that sends an AJAX request. Moreover, using SSL does not prevent a CSRF attack, because the malicious site can send an “[https://](https://)” request.
```

Comment:
a script that automatically submits the form or sends a form submission as an ajax request. The form could also be hidden using CSS so the user never realizes it's present.